redisinsight/ui/src/components/init/AppInit.tsx

```diff
@@ -1,16 +1,40 @@
-import React, { ReactElement, useEffect } from 'react'
+import React, { ReactElement, useCallback, useEffect } from 'react'
 import { useDispatch, useSelector } from 'react-redux'
-import { appInitSelector, initializeAppAction, STATUS_SUCCESS } from 'uiSrc/slices/app/init'
+import {
+  appInitSelector,
+  initializeAppAction,
+  STATUS_FAIL,
+  STATUS_INITIAL,
+  STATUS_SUCCESS,
+} from 'uiSrc/slices/app/init'
+import ConnectivityError from 'uiSrc/components/connectivity-error/ConnectivityError'
 import PagePlaceholder from '../page-placeholder'
 
-const AppInit = ({ children }: { children: ReactElement }) => {
+type Props = {
+  children: ReactElement
+  onSuccess?: () => void,
+  onFail?: () => void,
+}
+
+const AppInit = ({ children, onSuccess, onFail }: Props) => {
   const dispatch = useDispatch()
-  const { status } = useSelector(appInitSelector)
+  const {
+    status,
+    error = 'Something went wrong, please try again later',
+  } = useSelector(appInitSelector)
+
+  const initApp = useCallback(() => dispatch(initializeAppAction(onSuccess, onFail)), [])
 
   useEffect(() => {
-    dispatch(initializeAppAction())
+    if (status === STATUS_INITIAL) {
+      initApp()
+    }
   }, [])
 
+  if (status === STATUS_FAIL) {
+    return <ConnectivityError isLoading={false} onRetry={initApp} error={error} />
+  }
+
   return status === STATUS_SUCCESS ? children : <PagePlaceholder />
```

Comment:
Instead of `<PagePlaceholder />`, I think we need to render `<SuspenseLoader />`. Currently, it just renders a black screen while initializing. It'd be a better user experience to show a loader while we're fetching data in case it takes a few seconds.

Reply:
I agree about the loader, current logic was simply following the Crsf component's logic

Reply:
Can we go ahead and change? It's an easy change and it's better than showing a black screen to the user.